src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -570,14 +572,15 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     timestamp?: number,
     onlyOneBlock: boolean = false
   ) => {
-    await this.#blockBeingSavedPromise;
     const nextBlock = this.#readyNextBlock(this.blocks.latest, timestamp);
+    const transactions = await this.#miner.mine(
+      nextBlock,
+      maxTransactions,
+      onlyOneBlock
+    );
+    await this.#blockBeingSavedPromise;
```

Comment:
Await _after_ mining so we don't return until mining is actually fully completed. I tested this against the convex shutdown repo and it did not show a performance decrease. But this would more have an impact if there is a test that sends many transactions rather than just one big one. I'm open to ways to further test for performance degredation.

Reply:
Regardless, I hope to follow this PR with one that will allow the miner to manage the queue of mining requests and when the blocks being saved actually needs to be awaited. I just wanted to get this fix in sooner so I can feel like I'm making progress :smile: